neuralpredictors/measures/np_functions.py

```diff
@@ -12,10 +14,157 @@ def corr(y1, y2, axis=-1, eps=1e-8, **kwargs):
+
+
+def explainable_var(repeated_outputs, eps=1e-9):
+    """
+    Compute the explainable variance per neuron.
+
+    Args:
+        repeated_outputs (array): numpy array with shape (images, repeats, responses), or a list of lists of
+            responses (with len: num of repeats) per image.
+
+    Returns:
+        array: Corrected oracle correlations per neuron
+    """
+
+    ImgVariance = []
+    TotalVar = np.var(np.vstack(repeated_outputs), axis=0, ddof=1)
+    for out in repeated_outputs:
+        ImgVariance.append(np.var(out, axis=0, ddof=1))
+    ImgVariance = np.vstack(ImgVariance)
+    NoiseVar = np.mean(ImgVariance, axis=0)
+    explainable_var = (TotalVar - NoiseVar) / (TotalVar + eps)
+    return explainable_var
+
+
+def fev(targets, predictions, return_exp_var=False):
+    """
+    Compute the fraction of explainable variance explained per neuron
+
+    Args:
+        targets (list): Neuronal responses (ground truth) to image repeats. Dimensions:
+            [num_images] np.array(num_repeats, num_neurons)
+        outputs (list): Model predictions to the repeated images, with an identical shape as the targets
+        return_exp_var (bool): returns the fraction of explainable variance per neuron if set to True
+    Returns:
+        FEVe (np.array): the fraction of explainable variance explained per neuron
+        --- optional: FEV (np.array): the fraction
+    """
+
+    ImgVariance = []
+    PredVariance = []
+    for i, _ in enumerate(targets):
+        PredVariance.append((targets[i] - predictions[i]) ** 2)
+        ImgVariance.append(np.var(targets[i], axis=0, ddof=1))
+    PredVariance = np.vstack(PredVariance)
+    ImgVariance = np.vstack(ImgVariance)
+
+    TotalVar = np.var(np.vstack(targets), axis=0, ddof=1)
+    NoiseVar = np.mean(ImgVariance, axis=0)
+    FEV = (TotalVar - NoiseVar) / TotalVar
+
+    PredVar = np.mean(PredVariance, axis=0)
+    FEVe = 1 - (PredVar - NoiseVar) / (TotalVar - NoiseVar)
+    return [FEV, FEVe] if return_exp_var else FEVe
+
+
+def snr(repeated_outputs, per_neuron=True):
+    """
+    Compute signal to noise ratio.
+
+    Args:
+        repeated_outputs (array-like): numpy array with shape (images, repeats, responses), or a list of lists of
+            responses (with len: num of repeats) per image.
+        per_neuron (bool, optional): Return snr per neuron or averaged across neurons. Defaults to True.
+
+    Returns:
+        array: Signal to noise ratio per neuron or averaged across neurons.
+    """
+    repeated_outputs = anscombe(repeated_outputs)
+    mu = np.array([np.mean(repeats, axis=0) for repeats in repeated_outputs])
+    mu_bar = np.mean(mu, axis=0)
+    sigma_2 = np.array([np.var(repeats, ddof=1, axis=0) for repeats in repeated_outputs])
+    sigma_2_bar = np.mean(sigma_2, axis=0)
+    snr = (1 / mu.shape[0] * np.sum((mu - mu_bar) ** 2, axis=0)) / sigma_2_bar
+    return snr if per_neuron else np.mean(snr)
+
+
+def anscombe(x):
```

Comment:
This is not the only place where Anscombe transformation occurs - let's decide on where to house this function.

Reply:
Following Konstantin L.'s suggestion I'll move it to neuralpredictors.utils